modules/cardinal_service_rest/src/Plugin/rest/resource/TermsUsedResource.php

```diff
@@ -0,0 +1,165 @@
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\FieldConfigInterface;
+use Drupal\rest\Plugin\ResourceBase;
+use Drupal\rest\ResourceResponse;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Rest endpoint to provide data about what entities are tagged with terms.
+ *
+ * @RestResource(
+ *   id = "terms_used_resource",
+ *   label = @Translation("Terms In Use Resource"),
+ *   uri_paths = {
+ *     "canonical" = "/api/terms-used/{node_type}"
+ *   }
+ * )
+ */
+class TermsUsedResource extends ResourceBase {
+
+  const ENTITY_TYPE = 'node';
+
+  /**
+   * Entity Type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Entity field manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $fieldManager;
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->getParameter('serializer.formats'),
+      $container->get('logger.factory')->get('rest'),
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager')
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, array $serializer_formats, LoggerInterface $logger, EntityTypeManagerInterface $entityTypeManager, EntityFieldManagerInterface $field_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $serializer_formats, $logger);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->fieldManager = $field_manager;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function permissions() {
+    return [];
+  }
+
+  /**
+   * Get request on the rest endpoint.
+   *
+   * @param string $node_type
+   *   Node type bundle id.
+   *
+   * @return \Drupal\rest\ResourceResponse
+   *   Rest response.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function get($node_type) {
+    $data = [];
+
+    $fields = $this->fieldManager->getFieldDefinitions(self::ENTITY_TYPE, $node_type);
+    foreach ($fields as $field_name => $field_definition) {
+      if (
+        $field_definition instanceof FieldConfig &&
+        $field_definition->getType() == 'entity_reference' &&
+        $field_definition->getSetting('handler') == 'default:taxonomy_term'
+      ) {
+        $data[$field_name] = $this->getFieldTermsData($field_definition);
+      }
+    }
+
+    $data = array_filter($data);
```

Comment:
Will the $fields var ever not be a valid array of field definitions? Same with $data var in line 101 -- will this ever be empty or not an array?

Reply:
in line 90, the method `getFieldDefinitions` on the service will always return an array of [field definition interfaces](https://git.drupalcode.org/project/drupal/-/blob/9.0.x/core/lib/Drupal/Core/Entity/EntityFieldManagerInterface.php#L42).
And `$data` will always be some form of array since it's being set on line 88.